stingray/bispectrum.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import division
+import numpy as np
+from scipy.linalg import hankel
+import scipy.fftpack
+import stingray.lightcurve as lightcurve
+import matplotlib.pyplot as plt
+
+# import stingray.utils as utils
+# from stingray.utils import simon
+
+__all__ = ["Bispectrum"]
+
+
+class Bispectrum(object):
+
+    def __init__(self, lc=None, m=64, nfft=128):
+        """
+        Parameters
+        ----------
+        lc: lightcurve.Lightcurve object, optional, default None
+        The light curve data to be Fourier-transformed.
+
+        m : Number of samples in each segment Default :64(expecting huge data)
+
+        nfft : Fft size Default is 128
+
+
+        attributes
+        ----------
+        k: Number of segments
+
+        m : Number of samples in each segment Default :64(expecting huge data)
+
+        nfft : Fft size Default is 128
+
+        nsamp  : total numberof samples from the input signal
+
+        """
+
+        if lc is not None:
+            pass
+        else:
+            self.k = None
+            self.nfft = nfft
+            self.m = m
+            self.nsamp = None
+            return
+        self.m = m
+        self.nfft = nfft
+        assert isinstance(lc, lightcurve.Lightcurve), \
+            "lc must be a lightcurve.Lightcurve object!"
+
+        lc.counts = lc.counts.ravel(order='C')
+        self.nsamp = lc.counts.shape[0]
+
+        assert isinstance(m, int), "m is not an int !"
+
+        assert isinstance(nfft, int), "nfft is not an int !"
+
+        self.k = self.nsamp // self.m
```

Comment:
I'm wondering whether there shouldn't be a call to `compute_bispectrum` in `__init__`, since the whole point of this class is to compute a bispectrum. I'd make `compute_bispectrum` a private method, and call it in `__init__` to save an attribute `bispec` or something descriptive? This would be in line with what `Powerspectrum` and `Lightcurve` do.

This would also fix the comment further below about having to compute the bispectrum before being able to plot it.